### PR TITLE
Simplifies Subsequent Mask 

### PR DESCRIPTION
**Pull request by vidurj**
The subsequent_mask function in the bidirectional_language_model_transformer seemed to be unnecessarily convoluted.

### CODE REVIEW

File: allennlp/modules/seq2seq_encoders/bidirectional_language_model_transformer.py [hunk 1]
```diff
@@ -39,10 +39,8 @@ def attention(query: torch.Tensor,
 
 def subsequent_mask(size: int, device: str = 'cpu') -> torch.Tensor:
     """Mask out subsequent positions."""
-    attn_shape = (1, size, size)
-    mask = np.triu(np.ones(attn_shape), k=1).astype('uint8')
-    mask = (torch.from_numpy(mask) == 0)
-    return mask.to(device)
+    mask = torch.tril(torch.ones(size, size, device=device)).unsqueeze(0)
```
Comment (brendan-ai2) [inline allennlp/modules/seq2seq_encoders/bidirectional_language_model_transformer.py]: Shouldn't the type of `uint8` be preserved?

Comment (vidurj) [reply allennlp/modules/seq2seq_encoders/bidirectional_language_model_transformer.py]: The old return type was bool because of `torch.from_numpy(mask) == 0`. At the only place this is being used in the code base currently we have `subsequent = subsequent_mask(timesteps, device).int()` so the return type doesn't matter. i could cast to `int` to match this use case if that's preferable.

**Comment by brendan-ai2** [comment]
Thanks for creating things as int directly. Just needs a small lint fix: `W: 15, 0: Unused numpy imported as np (unused-import)`

**Comment by vidurj** [comment]
Thanks. Merging.
